_posts/2025-05-09-opentelemetry-zero-code-instrumentation.md

```diff
@@ -0,0 +1,537 @@
+---
+title: "Zero-code Instrumentation for OpenFaaS Functions with OpenTelemetry"
+description: "Learn how to capture telemetry data for OpenFaaS functions without modifying your code."
+date: 2025-05-09
+author_staff_member: han
+categories:
+- functions
+- otel
+- traces
+dark_background: true
+image: ""
+hide_header_image: true
+---
+
+Zero-code instrumentation for OpenFaaS function with OpenTelemetry
```

Comment:
In the introduction explain that the article will show the reader how to gather traces using OpenTelemetry. Explain that tracing can be done manually by adding code to the function, or for some languages using auto-instrumentation.

Link to the docs on auto-instrumentation for OTEL.

`Why would you instrument OpenFaaS functions?`

Good, but change to `What value does tracing bring to functions?`

Cover the cons along with the pros - additional infrastructure, cost, and mental load.

Mention that the article only covers tracing, but OpenTelemetry aims to be a comprehensive solution to cover traces, logs, and metrics.